Architecture_Engine/Query/Distance.cs

```diff
@@ -0,0 +1,90 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Architecture.BuildersWork;
+using BH.oM.Architecture.Elements;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Spatial.ShapeProfiles;
+using BH.Engine.Geometry;
+using BH.oM.Geometry.CoordinateSystem;
+
+namespace BH.Engine.Architecture
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static double Distance(this Opening opening1, Opening opening2, double maxExpectedDistance = double.NaN, double distanceTolerance = Tolerance.Distance, double angleTolerance = Tolerance.Angle)
+        {
+
+            if (opening1 == null || opening2 == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Opening is null");
+                return double.NaN;
+            }
+
+            double minDistance = double.MaxValue;
+
+            Point centerPoint2 = opening2.CoordinateSystem.Origin;
+            Plane plane2 = new Plane { Origin = centerPoint2, Normal = opening2.CoordinateSystem.Z };
+
+            if (!opening1.IsValidForDistanceCheck(opening2, maxExpectedDistance, distanceTolerance, angleTolerance)) 
+            {
+                BH.Engine.Reflection.Compute.RecordError("Distance can't be calculated");
+                return double.NaN;
+            }
+
+            else
+            {               
+                foreach (ICurve edge1 in opening1.Profile.Edges)
+                {
+                    foreach (ICurve edge2 in opening2.Profile.Edges)
+                    {
+                        Cartesian global = new Cartesian();
+                        ICurve edge1O = edge1.Orient(global, opening1.CoordinateSystem);
+                        ICurve edge1Onew = edge1O.IProject(plane2);
+                        ICurve edge2O = edge2.Orient(global, opening2.CoordinateSystem);
+
+                        double dist = edge1Onew.Distance(edge2O);
+                        if (dist < minDistance)
+                        {
+                            minDistance = dist;
+                        }
```

Comment:
```suggestion
                        minDistance = Math.Min(minDistance, dist);
```

Not sure how this fits in with @pawelbaran above comment but this is a slightly neater way of ensuring the minimum distance is always the smaller of the two variables that is quite common place in various parts of the code. As this isn't my engine, this is not a mandatory suggestion if you prefer to keep the `if` statement that's fine too :smile;